src/Lecture3.hs

```diff
@@ -213,8 +230,24 @@ types that can have such an instance.
 -- instance Foldable Weekday where
 -- instance Foldable Gold where
 -- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
+
+instance Foldable List1 where
+  foldr :: (a -> b -> b) -> b -> List1 a -> b
+  foldr f z (List1 x []) = f x z
```

Comment:
This case is redundant as it's already covered by the next line
```suggestion
```

Reply:
Indeed... It seems to also be the case for foldMap and fmap for List1. Unless there is a reason to keep these cases in those functions ?

Reply:
In this cases, you don't need to handle empty lists separately 🆗